### PR TITLE
Make java fixtures snippet CC compatible

### LINKED ISSUE #21817: CCH22-20: Adjust the `snippets/java/fixtures` sample for configuration cache
The sample project located under [`subprojects/docs/src/snippets/java/fixtures`](https://github.com/gradle/gradle/tree/master/subprojects/docs/src/snippets/java/fixtures) does not meet [the configuration cache requirements](https://docs.gradle.org/nightly/userguide/configuration_cache.html#config_cache:requirements).

The build scripts in the `groovy` and `kotlin` parts of the sample project need to be adjusted to follow the requirements.
### Context
This issue is a part of the [Gradle Configuration Cache Hackathon 2022](https://github.com/gradle/cc-hackathon-2022).

If you wish to participate and contribute the fix, please follow [the suggested workflow](https://github.com/gradle/cc-hackathon-2022/blob/main/workflow.md).
### Expected behavior

- The sample project is fixed so that the related documentation tests pass with configuration cache enabled.
- The tests related to this sample project are removed from the `testsToBeFixedForConfigurationCache` list in `subprojects/docs/build.gradle`. You can find the test IDs below.
- The doc pages that include the snippets from this sample project might need to be updated to be consistent with the sample. Feel free to ask the team for help with this! You can find the pages listed below.
### Test failure details
These tests running in `:docs:docsTest` check the sample project and are failing with configuration cache enabled:
* `snippet-java-fixtures_groovy_javaTestFixtures.sample`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket2SnippetsTest/snippet-java-fixtures_groovy_javaTestFixtures.sample)
    ```
    Could not get unknown property 'components' for task ':lib:usages' of type org.gradle.api.DefaultTask.
    ```

* `snippet-java-fixtures_kotlin_javaTestFixtures.sample`[(Build Scan)](https://ge.gradle.org/s/wkszl3ueqiln2/tests/:docs:docsTest/org.gradle.docs.samples.Bucket2SnippetsTest/snippet-java-fixtures_kotlin_javaTestFixtures.sample)
    ```
    Task `:lib:usages` of type `org.gradle.api.DefaultTask`: cannot serialize Gradle script object references as these are not supported with the configuration cache.
    ```

### How to reproduce
To reproduce the failure in the Gradle documentation tests, run the following in the `gradle/gradle`repository:
```sh
./gradlew :docs:docsTest \
    -PrunBrokenConfigurationCacheDocsTests=true \
    -PenableConfigurationCacheForDocsTests=true \
    --tests "*snippet-java-fixtures_groovy_javaTestFixtures.sample" \
    --tests "*snippet-java-fixtures_kotlin_javaTestFixtures.sample"
```
In the test build outputs, which you can find in the test report, the configuration cache problems are listed along with the URL pointing to the detailed HTML problems report that can help investigating the violation. 
You can find more instructions on working with the sample projects in the [FAQ](https://github.com/gradle/cc-hackathon-2022/blob/main/faq.md).
### Docs including the snippets from this sample project
The following doc pages include the snippets from this sample project: 
- `subprojects/docs/src/docs/userguide/jvm/java_testing.adoc`;

### PR DESCRIPTION
**Pull request by 3flex**
Resolves component usages at configuration time

Fixes: #21817

### CODE REVIEW

**Review by nathan-contino — APPROVED** [review]
Docs LGTM

File: subprojects/docs/src/snippets/java/fixtures/groovy/lib/build.gradle [hunk 1]
```diff
@@ -46,7 +46,8 @@ components.java.withVariantsFromConfiguration(configurations.testFixturesRuntime
 // end::disable-test-fixtures-publishing[]
 
 tasks.create("usages") {
+    def javaComponent = components.java.usages.collect { it.name }
```
Comment (abstratt) [inline subprojects/docs/src/snippets/java/fixtures/groovy/lib/build.gradle:49]: Thanks for this @3flex. Could you please a more accurate name for the local variable (`variantNames`, `usages` etc)? (here and in the kotlin version)

**Review by abstratt — CHANGES_REQUESTED** [review]

**Comment by abstratt** [comment]
@bot-gradle test this

File: subprojects/docs/src/snippets/java/fixtures/groovy/lib/build.gradle (hunk 1 above)
Comment (3flex) [reply subprojects/docs/src/snippets/java/fixtures/groovy/lib/build.gradle]: Apologies, that was sloppy! I went with `javaComponentUsages`, hope that's appropriate.

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Comment by abstratt** [comment]
@bot-gradle test and merge

**Review by abstratt — APPROVED** [review]
Thanks!

**Comment by abstratt** [comment]
🚀

**Pull request merged**